pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -158,6 +155,31 @@ func warnDeprecatedAPIs(spec *v1.VirtualMachineInstanceSpec, config *virtconfig.
 	return warnings
 }
 
+func ValidateVirtualMachineInstancePerArch(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec, config *virtconfig.ClusterConfig) []metav1.StatusCause {
+	var causes []metav1.StatusCause
+
+	switch {
+	case webhooks.IsAMD64(spec, config):
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceAmd64Setting(field, spec)...)
+	case webhooks.IsS390X(spec, config):
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceS390XSetting(field, spec)...)
+	case webhooks.IsARM64(spec, config):
+		causes = append(causes, webhooks.ValidateVirtualMachineInstanceArm64Setting(field, spec)...)
+	default:
```

Comment:
nit: I think it would do the following:
```suggestion
	switch spec.Architecture {
	case "arm64":
		causes = append(causes, webhooks.ValidateVirtualMachineInstanceAmd64Setting(field, spec)...)
	case "s390x:
		causes = append(causes, webhooks.ValidateVirtualMachineInstanceS390XSetting(field, spec)...)
	case "arm64":
		causes = append(causes, webhooks.ValidateVirtualMachineInstanceArm64Setting(field, spec)...)
	default:
```

Then raise an error. Then, you can remove all of the helper functions which AFAICT aren't being used anywhere else.

I'd be happy to see this, but it shouldn't block your PR.